### PR TITLE
test randomize with error

### PR DESCRIPTION
**Pull request by rob-solana**
#### Problem
 unlock needs to be in order, because we mix lock and exec errors together

 #### Summary of Changes
 * add test that errors don't cause dangling locks
 * remove entry size restrictions, related code, blob tests

Fixes #

### CODE REVIEW

**Comment by rob-solana** [comment]
CC @mvines

File: core/src/blocktree_processor.rs [hunk 1]
```diff
@@ -1253,6 +1253,57 @@ pub mod tests {
...
+                            &keypairs[i + j],
+                            &keypairs[i + j + NUM_TRANSFERS].pubkey(),
+                            1,
+                            bank.last_blockhash(),
+                        )
+                    })
+                    .collect::<Vec<_>>();
+
+                transactions.push(system_transaction::create_account(
+                    &mint_keypair,
```
Comment (rob-solana) [inline core/src/blocktree_processor.rs]: net result of this array of entries is that (at least) mint_keypair stays locked

File: core/src/blocktree_processor.rs [hunk 2]
```diff
@@ -1230,6 +1230,57 @@ pub mod tests {
...
+                    bank.last_blockhash(),
+                    1,
+                    0,
+                    &Pubkey::new_rand(),
+                ));
+
+                next_entry_mut(&mut hash, 0, transactions)
+            })
+            .collect();
+        assert_eq!(process_entries(&bank, &entries, true), Ok(()));
```
Comment (carllin) [inline core/src/blocktree_processor.rs]: Won't a transaction error cause the return value of `process_entries` to be an error

Comment (rob-solana) [reply core/src/blocktree_processor.rs]: no, because this kind of transaction failure occurs in the instruction processor, ends up in the ledger

**Pull request merged**